src/Backend/Modules/Profiles/Actions/Delete.php

```diff
@@ -32,37 +32,37 @@ public function execute(): void
         $this->id = (int) $deleteFormData['id'];
 
         // does the item exist
-        if ($this->id !== 0 && BackendProfilesModel::exists($this->id)) {
-            // call parent, this will probably add some general CSS/JS or other required files
-            parent::execute();
+        if ($this->id === 0 || !BackendProfilesModel::exists($this->id)) {
+            $this->redirect(BackendModel::createURLForAction('Index') . '&error=non-existing');
+        }
 
-            // get profile
-            $profile = BackendProfilesModel::get($this->id);
+        // call parent, this will probably add some general CSS/JS or other required files
+        parent::execute();
 
-            // already deleted? Prolly want to undo then
-            if ($profile['status'] === 'deleted') {
-                // set profile status to active
-                BackendProfilesModel::update($this->id, ['status' => 'active']);
+        // get profile
+        $profile = BackendProfilesModel::get($this->id);
 
-                // redirect
-                $this->redirect(
-                    BackendModel::createURLForAction('Index') . '&report=profile-undeleted&var=' . rawurlencode(
-                        $profile['email']
-                    ) . '&highlight=row-' . $profile['id']
-                );
-            } else {
-                // delete profile
-                BackendProfilesModel::delete($this->id);
+        // already deleted? Prolly want to undo then
+        if ($profile['status'] === 'deleted') {
+            // set profile status to active
+            BackendProfilesModel::update($this->id, ['status' => 'active']);
 
-                // redirect
-                $this->redirect(
-                    BackendModel::createURLForAction('Index') . '&report=profile-deleted&var=' . rawurlencode(
-                        $profile['email']
-                    ) . '&highlight=row-' . $profile['id']
-                );
-            }
+            // redirect
+            $this->redirect(
+                BackendModel::createURLForAction('Index') . '&report=profile-undeleted&var=' . rawurlencode(
+                    $profile['email']
+                ) . '&highlight=row-' . $profile['id']
+            );
         } else {
```

Comment:
useless else since you redirect, but add a return to make it clear that it is the end